cmd/api-headers.go

```diff
@@ -166,6 +166,10 @@ func setObjectHeaders(w http.ResponseWriter, objInfo ObjectInfo, rs *HTTPRangeSp
 
 	if rs == nil && opts.PartNumber > 0 {
 		rs = partNumberToRangeSpec(objInfo, opts.PartNumber)
+		start, rangeLen, err = rs.GetOffsetLength(totalObjectSize)
```

Comment:
I believe it is just easier if you move line 167 before line 162

Reply:
How is it easier?